src/content/resources/cpp/multidimensional-arrays.md

```diff
@@ -0,0 +1,122 @@
+So there are 2 arrays with 3 elements each within `arr`.
+
+##### Method 2:
+Use: 
+```c++
+data_type array_name[][] = {{a1,a2,a3,...},{b1,b2,b3,...}};
+``` 
+Where `data_type` is an data type(such as `int`, `char` or etc.), `array_name` is what the name of the array will be and `a1, a2, a3, b1, b2, b3,...` are objects of the data type that was chosen at the start in `data_type`, but the number of `a`'s and `b`'s must be equal.
+
+Example 2: 
+```c++
+int arr[][] = {{1,2,3},{4,5,6}};
+``` 
+This will make a 2x3 matrix with preset values instead of all 0's so it'll look like: 
+```c++
+{{1,2,3},{4,5,6}}
+```
+
+##### Method 3:
+
+Combine both: 
+```c++
+data_type array_name[A][B] = {{a1,a2,a3,...},{b1,b2,b3,...}}
+``` 
+Where `data_type` is an data type(such as `int`, `char` or etc.), `array_name` is what the name of the array will be and `a1, a2, a3, b1, b2, b3,...` are objects of the data type that was chosen at the start in `data_type` and `A` and `B` is the number of `{}`'s and `a`'s and `b`'s respectively.
+
+Example 3: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+```
+This will make a 2x3 matrix that looks like: 
+```c++
+{{1,2,3},{4,5,6}}
+```
+
+##### Note:
+It is possible to go ever further, and so a declaration like `data_type array_name[A][B][C]` is possible and would add more layers to your array. So after going through one layer, you have to input the location of the next layer, and so on and so forth until you reach the deepest layer of the `{}` where you'll find the object. 
+
+Example 4: 
+```c++
+int arr[2][3][4];
+``` 
+This will make an 2x3x4 matrix that looks like: 
+```c++
+{{{0,0,0,0},{0,0,0,0},{0,0,0,0}},{{0,0,0,0},{0,0,0,0},{0,0,0,0}}}
+```
+
+### What can you do with it?
+
+##### Calling objects
+To get the object inside, you can call for the object inside a multidimensional array by putting a integer inside the `[]`.
+
+So taking the array made in example 2 above: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+std::cout << arr[0][1]; 
+```
+This would output `2` because using array indicies, `2` is the object in position 1 of the 0th array.
+
+You must put an integer in both `[]`, or else you will get an error. 
+There is no syntax to instantly view the whole array, but an easy way is to loop through the whole array and outputting it.  
+
+##### Reassigning objects
+Like with a regular array, you can also reassign what object is being held inside a exact position using `=`
+So let's take for example the array made in example 2 above, which looks like: 
+```c++
+int arr[2][3] = {{1,2,3}, {4,5,6}};
+arr[1][1] = 1337;
+```
+This would cause the array to change to looking like:
+```c++
+{{1,2,3},{4,1337,6}}
+```
+
+### Restrictions
+
+1. The size is locked upon initialization. Once you make the array, its size is locked. For example, if you make a 2x3 array, and realize you need another column or row, you have to either make a new one, or change the size where you initialize it.
```

Comment:
Could probably mention that if you want a dynamic sized array you can opt for `std::vector`

Reply:
Okay, I added that, but I feel like explaining in-depth how should be in an article for `std::vector`